### PR TITLE
[PATCH v4] linux-gen: packet: inline misc fast path functions

### PR DESCRIPTION
**Pull request by MatiasElo**

### CODE REVIEW

File: platform/linux-generic/include/odp/api/plat/packet_inlines.h [hunk 1]
```diff
@@ -369,6 +371,22 @@ _ODP_INLINE odp_packet_chksum_status_t odp_packet_l4_chksum_status(odp_packet_t
 	return ODP_PACKET_CHKSUM_OK;
 }
 
+_ODP_INLINE void odp_packet_l3_chksum_insert(odp_packet_t pkt, int insert)
+{
+	_odp_packet_flags_t *flags = _odp_pkt_get_ptr(pkt, _odp_packet_flags_t, flags);
```
Comment (TuomasTaipale) [inline platform/linux-generic/include/odp/api/plat/packet_inlines.h:431]: Perhaps this declare-and-define-in-the-same-line could be used in the earlier commits as well.

File: platform/linux-generic/include/odp/api/plat/packet_inlines.h [hunk 2]
```diff
@@ -433,6 +434,18 @@ _ODP_INLINE int8_t odp_packet_shaper_len_adjust(odp_packet_t pkt)
...
+_ODP_INLINE uint64_t odp_packet_cls_mark(odp_packet_t pkt)
+{
+	_odp_packet_input_flags_t input_flags;
+
+	input_flags.all = _odp_pkt_get(pkt, uint64_t, input_flags);
+
+	if (input_flags.cls_mark)
+		return _odp_pkt_get(pkt, uint16_t, cls_mark);
+
+	return 0;
```
[the remark this answers is not in the document]
Comment (MatiasElo) [reply platform/linux-generic/include/odp/api/plat/packet_inlines.h:621]: Changed in V2.

File: platform/linux-generic/include/odp/api/plat/packet_inlines.h [hunk 3]
```diff
@@ -274,6 +277,62 @@ _ODP_INLINE void *odp_packet_l4_ptr(odp_packet_t pkt, uint32_t *len)
...
+	return ODP_PROTO_L2_TYPE_NONE;
+}
+
+_ODP_INLINE odp_proto_l3_type_t odp_packet_l3_type(odp_packet_t pkt)
+{
+	_odp_packet_input_flags_t input_flags;
+
+	input_flags.all = _odp_pkt_get(pkt, uint64_t, input_flags);
+
+	if (input_flags.ipv4)
```
[the remark this answers is not in the document]
Comment (MatiasElo) [reply platform/linux-generic/include/odp/api/plat/packet_inlines.h:357]: I'd prefer not to do major changes to the function implementations as the target of this PR is inlining.

File: platform/linux-generic/include/odp/api/plat/packet_inlines.h (hunk 1 above)
Comment (MatiasElo) [reply platform/linux-generic/include/odp/api/plat/packet_inlines.h:431]: The difference is that here we take a pointer to the packet flags. In the previous commits I followed the style used in all previous flag accessor functions.

**Review by TuomasTaipale — APPROVED** [review]

**Comment by MatiasElo** [comment]
V3:
- Added reviewed-by tags

**Review by psavol — APPROVED** [review]

**Pull request merged**